src/algebra/polynomial/basic.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson, Jalex Stark.
+-/
+import data.polynomial
+open polynomial finset
+
+/-!
+# Polynomials
+
+Eventually, much of data/polynomial.lean should land here.
+
+## Main results
+
+We define two `ring_hom`s of type `polynomial R →+* R`,
+- `eval_ring_hom`
+- `coeff_zero_ring_hom`
+
+We define a `monoid_hom` of type `polynomial R →* R`,
+- `leading_coeff_monoid_hom`
+-/
+
+universes u w
+
+noncomputable theory
+
+variables {R : Type u} {α : Type w}
+
+namespace polynomial
+
+section comm_semiring
+variables [comm_semiring R]
+
+/-- A `ring_hom` which evaluates polynomials at a particular value -/
+def eval_ring_hom : R → (polynomial R →+* R) := eval₂_ring_hom (ring_hom.id R)
+
+@[simp]
+lemma coe_eval_ring_hom (r : R) (p : polynomial R) : eval_ring_hom r p = eval r p := rfl
```

Comment:
`coe_eval_ring_hom` should be about
`(eval_ring_hom r : polynomial R → R) = eval r`,
the lemma you wrote is `eval_ring_hom_apply`.
Both are useful.

Reply:
This got deleted